include/forcing/NullForcingProvider.hpp

```diff
@@ -0,0 +1,271 @@
+        return available_forcings;
+    }
+
+    private:
+
+    /**
+     * @brief Checks forcing vector index bounds and adjusts index if out of vector bounds
+     * /// \todo: Bounds checking is based on precipitation vector. Consider potential for vectors of different sizes and indices.
+     */
+    inline void check_forcing_vector_index_bounds()
+    {
+        //Check if forcing index is less than zero and if so, set to zero.
+        if (forcing_vector_index < 0)
+        {
+            forcing_vector_index = 0;
+            /// \todo: Return appropriate warning
+            std::cout << "WARNING: Forcing vector index is less than zero. Therefore, setting index to zero." << std::endl;
+        }
+
+        //Check if forcing index is greater than or equal to the size of the size of the time vector and if so, set to zero.
+        else if (forcing_vector_index >= time_epoch_vector.size())
+        {
+            forcing_vector_index = time_epoch_vector.size() - 1;
+            /// \todo: Return appropriate warning
+            std::cout << "WARNING: Reached beyond the size of the forcing vector. Therefore, setting index to last value of the vector." << std::endl;
+        }
+
+        return;
+    }
+
+    /**
+     * Get the current value of a forcing param identified by its name.
+     *
+     * @param name The name of the forcing param for which the current value is desired.
+     * @param index The index of the desired forcing time step from which to obtain the value.
+     * @return The particular param's value at the given forcing time step.
+     */
+    inline double get_value_for_param_name(const std::string& name, int index) {
+        if (index >= time_epoch_vector.size() ) {
+            throw std::out_of_range("Forcing had bad index " + std::to_string(index) + " for value lookup of " + name);
+        }
+
+        std::string can_name = name;
+        if(data_access::WellKnownFields.count(can_name) > 0){
+            auto t = data_access::WellKnownFields.find(can_name)->second;
+            can_name = std::get<0>(t);
+        }
+
+        if (forcing_vectors.count(can_name) > 0) {
+            return forcing_vectors[can_name].at(index);
+        }
+        else {
+            throw std::runtime_error("Cannot get forcing value for unrecognized parameter name '" + name + "'.");
+        }
+    }
+
+    /**
+     * @brief Read Forcing Data from CSV
+     * Reads only data within the specified model start and end date-times.
+     * @param file_name Forcing file name
+     */
+    // No file to read
+    void read_csv()
+    {
+                //std::string var_name = CSDMS_STD_NAME_RAIN_VOLUME_FLUX;
+                available_forcings.push_back("time");
+                std::string var_name = "precip";
+                std::string units = "";
+                auto wkf = data_access::WellKnownFields.find(var_name);
+                if(wkf != data_access::WellKnownFields.end()){
```

Comment:
Actually, are either `time` or `precip` found in WellKnownFields? I think not (AorcForcing.hpp)... so available_forcings probably _is_ empty and this whole block is doing nothing. Right?

Reply:
No, by ```precip``` I meant to use ```precip_rate```. No sure if ```time``` is needed. I am trying to mimic a forcing file with some variable names here.

Reply:
So, if this provider returns _any_ variables, it should _not_ return one of the standard/AORC ones, because that will just interfere with (or at least confuse) getting the same variables from a BMI forcing provider. If we have to return some variable, it should be named `zero` or something as discussed in #569 .

Reply:
You are right. This function now has only one line of codes.